be/src/runtime/local_pass_through_buffer.h

```diff
@@ -30,29 +30,47 @@ class PassThroughChunkBuffer {
     PassThroughChunkBuffer(const TUniqueId& query_id);
     ~PassThroughChunkBuffer();
     PassThroughChannel* get_or_create_channel(const Key& key);
+    int ref() {
+        _ref_count += 1;
+        return _ref_count;
+    }
+    int unref() {
+        _ref_count -= 1;
+        return _ref_count;
+    }
 
 private:
     std::mutex _mutex;
     const TUniqueId _query_id;
     std::unordered_map<Key, PassThroughChannel*, KeyHash> _key_to_channel;
+    int _ref_count;
 };
-using PassThroughChunkBufferPtr = std::shared_ptr<PassThroughChunkBuffer>;
 
 class PassThroughContext {
 public:
-    PassThroughContext(const PassThroughChunkBufferPtr& chunk_buffer, TUniqueId fragment_instance_id,
-                       PlanNodeId node_id)
+    PassThroughContext(PassThroughChunkBuffer* chunk_buffer, TUniqueId fragment_instance_id, PlanNodeId node_id)
             : _chunk_buffer(chunk_buffer), _fragment_instance_id(fragment_instance_id), _node_id(node_id) {}
     void init();
     void append_chunk(int sender_id, const vectorized::Chunk* chunk, size_t chunk_size);
     void pull_chunks(int sender_id, ChunkUniquePtrVector* chunks, std::vector<size_t>* bytes);
 
 private:
     // hold this chunk buffer to avoid early deallocation.
-    PassThroughChunkBufferPtr _chunk_buffer;
+    PassThroughChunkBuffer* _chunk_buffer;
     TUniqueId _fragment_instance_id;
     PlanNodeId _node_id;
     PassThroughChannel* _channel = nullptr;
 };
 
+class PassThroughChunkBufferManager {
+public:
+    void open(const TUniqueId& query_id);
```

Comment:
comment one fragment call it once or one query call it one?

Reply:
OK, will do that.